loadfile/loadfile.go

```diff
@@ -0,0 +1,47 @@
+// Package loadfile provides functions to load files from a directory.
+package loadfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// LoadContext reads the contents at each file into a map where the key
+// is the absolute filepath and file contents is the value.
+func LoadContext(neededFilepaths []string) (map[string][]byte, error) {
+	result := map[string][]byte{}
+	var err error
+	for _, filePath := range neededFilepaths {
+		absPath, err := filepath.Abs(filePath)
+		if err != nil {
+			return nil, fmt.Errorf("error loading context files, failed to obtain absolute path of file %s (%w)",
+				filepath.Base(filePath), err)
+		}
+		fileData, err := os.ReadFile(absPath) //nolint:gosec
```

Comment:
What lint complaint are you squelching here?

Reply:
From [gosec's readme](https://github.com/securego/gosec?tab=readme-ov-file#usage), `G304: File path provided as taint input`.

```shell
❯ golangci-lint run
loadfile/loadfile.go:21:20: G304: Potential file inclusion via variable (gosec)
		fileData, err := os.ReadFile(absPath)
```

I guess these are the same thing.

Reply:
further investigation, gosec's [G304 page](https://securego.io/docs/rules/g304.html#the-right-way) recommends a clean, safe path. I added a clean path, but I don't know how to incorporate a "safe" path, or if the "safe" part is necessary since adding `filepath.Clean()` satisfied the linter.

Reply:
Wow...that's an impressive level of paranoia...but, I suppose it's good.

However, I think you've found a bug in the linter, if all you needed to do was to add `Clean()` to the result returned by `Abs()`, because "[Abs calls Clean on the result](https://pkg.go.dev/path/filepath#Abs)".

(I concur that establishing "safety" is a bit of a challenge, so perhaps it is not necessarily required.)

Anyway, I originally asked the question because it's generally an anit-pattern to disable large chunks of the linter, especially without explanation.  (In fact, apparently there's a [lint check](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/nolintlint/README.md) for that....)  So, it would have been better to disable just `G304` rather than all of `gosec` (which might have allowed me to answer my own question), but, of course, fixing the code instead of disabling the linter is generally an even better solution!